rust/chains/hyperlane-ethereum/src/fallback.rs

```diff
@@ -0,0 +1,133 @@
+use std::fmt::Debug;
+
+use async_trait::async_trait;
+use ethers::providers::{Http, HttpClientError, JsonRpcClient, ProviderError};
+use serde::{de::DeserializeOwned, Serialize};
+use serde_json::Value;
+use thiserror::Error;
+use tracing::warn;
+
+use ethers_prometheus::json_rpc_client::PrometheusJsonRpcClient;
+
+const METHODS_TO_NOT_TO_FALLBACK_ON: &[&str] = &[
+    "eth_estimateGas",
+    "eth_sendTransaction",
```

Comment:
I think `eth_sendTransaction` and `eth_sendRawTransaction` may actually be worth falling back on

The reasoning is that with QuorumProvider the race condition is pretty clear where if we send a tx to a bunch of providers, if provider A quickly propagates that tx to provider B before provider B processes our RPC, then provider B will complain even though the tx was submitted successfully to the general network via provider A.

Then thinking through the fallback provider case:
1. submit to provider A, it gives us an error
2. We should then try submitting to provider B. There's the chance that provider A somehow erroneously gave us an error in (1), in which case provider B may already know our tx and also give an error, but I think this is much less likely than a case where provider A gives an error because it's just broken

wdyt?

Reply:
Well, this was actually the logic in retrying provider, not quorum provider. The reasoning was that if we got an error on this method it probably was a VM revert error and not a provider error.

Reply:
Lol https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/706#issuecomment-1177476366

I think my motivation was probably to do with errors from eth_sendTransaction / eth_sendRawTransaction to do with nonces - e.g. if we submitted a tx accidentally with nonce `n - 1` when the latest nonce on chain is `n`, we will get an error and don't want to retry. Fwiw VM reverts in the transaction won't result in errors from th_sendTransaction / eth_sendRawTransaction

But yeah good point that we did have a reason to add this to the retrying provider. I'm happy to keep as is where we don't fall back

Reply:
Okay, sg for now, might want to change in the future depending on what we observe.